pallets/did/src/lib.rs

```diff
@@ -240,6 +255,43 @@ where
 	}
 }
 
+/// A DID update request. It contains the following values:
+/// * The DID identifier being updated
+/// * The optional new authentication key to use
+/// * The optional new encryption key to use
+/// * The optional new attestation key to use
+/// * The optional new delegation key to use
+/// * The optional set of old attestation keys to remove
+/// * The optional new endpoint URL pointing to the DID service endpoints
+/// * A counter used to protect against replay attacks
+#[derive(Clone, Decode, Debug, Encode, PartialEq)]
+pub struct DidUpdateOperation<DidIdentifier>
+where
+	DidIdentifier: Parameter + Encode + Decode + Debug,
+{
+	did: DidIdentifier,
+	new_auth_key: Option<PublicVerificationKey>,
+	new_key_agreement_key: Option<PublicEncryptionKey>,
+	new_attestation_key: Option<PublicVerificationKey>,
+	new_delegation_key: Option<PublicVerificationKey>,
+	verification_keys_to_remove: Option<Vec<PublicVerificationKey>>,
```

Comment:
Same as above. Initially this was an `Option<BTreeSet<PublicVerificationKey>>`, but polkadot-js was only allowing me to input it as a HEX-encoded byte array, so there had to be something wrong in the types. Tests were running ok, and I would to revert to using a `BTreeSet` tho.